src/BizHawk.Client.EmuHawk/CoreDownloaderForm.cs

```diff
@@ -0,0 +1,108 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Drawing;
+using System.IO;
+using System.Net.Http;
+using System.Threading;
+using System.Threading.Tasks;
+using System.Windows.Forms;
+
+using BizHawk.Common;
+using BizHawk.Common.PathExtensions;
+
+namespace BizHawk.Client.EmuHawk
+{
+	public partial class CoreDownloaderForm : Form
+	{
+		// TODO edit
+		private static readonly string BaseUrl = $"https://github.com/Morilli/BizHawk-extra/raw/{VersionInfo.MainVersion}/";
+		private static readonly string OutputPath = Path.Combine(PathUtils.DataDirectoryPath, "dll");
```

Comment:
This is correct, but the dynamic loader for each core has to be changed to look here. (This is for NixOS where `BIZHAWK_HOME` and `BIZHAWK_DATA_HOME` aren't the same.)

Reply:
What is `BIZHAWK_DATA_HOME` intended to be used for? In the core I only see references from the ffmpeg and RAIntegration downloader dialogs, and the gamedb loading code. I'm assuming `PathUtils.DllDirectoryPath` would be the correct one to use here, but what IS supposed to be put in `PathUtils.DataDirectoryPath`? Are the existing downloader forms also outputting into an incorrect location?

Reply:
`PathUtils.DllDirectoryPath` is read-only on NixOS.
Ideally, these optional components would be built from source and installed at the same time as EmuHawk, but I haven't looked into that for FFmpeg. (I had no idea RAIntegration used it, but IIRC that's a Windows-only component?)

Reply:
Reading this again, is `BIZHAWK_DATA_HOME` shared between different versions of bizhawk? In that case this would actually pose a problem if non-static dlls were written to it.

Reply:
Out-of-the-box, it points to e.g. `~/.local/share/emuhawk-monort-2.9.1`, so it's shared between instances of the same version.
I think you should use `DllDirectoryPath` and block the downloader on Linux. Non-NixOS distros can ship a lite and full package if there's demand.